skbio/stats/distance/_regression.py

```diff
@@ -0,0 +1,170 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+
+import six
+import numpy as np
+from skbio.util._decorator import experimental
+from skbio.util._misc import check_random_state
+from skbio.stats.distance._mantel import _order_dms
+
+from skbio.stats.distance import DistanceMatrix
+
+@experimental(as_of="0.4.0")
+def linregress(y, *args, **kwargs):
```

Comment:
Thanks for looking into these implementations. I'm not opposed to making scikit-learn or statsmodels a dependency at some point but it doesn't appear to be worth it just for this one function IMO.

What about making `regress` private (renaming it `_regress`)? That way the code is easier to read, avoiding a long function containing an inner function. Then we won't have to support this as part of scikit-bio's public API and can swap it out for another implementation at any point in the future.

Also, :+1: to trying out scipy's functionality in `_regress` for improved performance and/or code reduction.